### PR TITLE
feat: handler expression error

### PR DESCRIPTION
**Pull request by MrWindlike**
- Displaying the expression value and error

https://user-images.githubusercontent.com/25414733/161890228-eac04830-b51f-4c8e-871e-ce20078c69e2.mov
- Improving the form performance

### CODE REVIEW

File: packages/editor/src/validator/rules/PropertiesRules.ts [hunk 1]
```diff
@@ -14,21 +16,17 @@ class PropertySchemaValidatorRule implements PropertiesValidatorRule {
     component,
     trait,
     validators,
+    stateManager
```
Comment (tanbowensg) [inline packages/editor/src/validator/rules/PropertiesRules.ts:19]: The validator is for static validation by design, so we can not add `stateManager` in validator.

Comment (MrWindlike) [reply packages/editor/src/validator/rules/PropertiesRules.ts:19]: Should we display the runtime expression errors in the waring area?

Comment (tanbowensg) [reply packages/editor/src/validator/rules/PropertiesRules.ts:19]: Maybe not now. We'd better focus on expression editor this time.

Comment (MrWindlike) [reply packages/editor/src/validator/rules/PropertiesRules.ts]: OK, I remove the codes.

File: packages/runtime/src/components/_internal/ImplWrapper.tsx [hunk 2]
```diff
@@ -154,7 +155,17 @@ const _ImplWrapper = React.forwardRef<HTMLDivElement, ImplWrapperProps>((props,
     };
   }, [c.id, stateManager.store]);
 
-  const mergedProps = { ...evaledComponentProperties, ...propsFromTraits };
+  const mergedProps = useMemo(() => {
+    const mergedProps: Record<string, any> = { ...evaledComponentProperties, ...propsFromTraits };
```
Comment (tanbowensg) [inline packages/runtime/src/components/_internal/ImplWrapper.tsx:159]: `mergedProps` is declared outside, use other name.

File: packages/editor-sdk/src/components/Widgets/ExpressionWidget.tsx [hunk 3]
```diff
@@ -132,17 +135,60 @@ export const ExpressionWidgetOptionsSchema = Type.Object({
...
+  const evalCode = useCallback(
+    (code: string) => {
+      try {
+        const result = services.stateManager.maskedEval(getParsedValue(code, type));
+
+        validate(result);
+        if (validate.errors?.length) {
+          const err = validate.errors[0];
+
+          if (err.message?.includes('should be')) {
```
Comment (tanbowensg) [inline packages/editor-sdk/src/components/Widgets/ExpressionWidget.tsx:163]: Is there other way to judge error type? This way is not reliable.

File: packages/runtime/src/components/_internal/ImplWrapper.tsx (hunk 2 above)
Comment (tanbowensg) [reply packages/runtime/src/components/_internal/ImplWrapper.tsx:159]: This needs fix.

Comment (MrWindlike) [reply packages/runtime/src/components/_internal/ImplWrapper.tsx]: Done.

**Review by tanbowensg — APPROVED** [review]
LGTM

**Pull request merged**